### PR TITLE
Added a validation for dob that is not very recent (2 years).

### LINKED ISSUE #3455: Reject very-recent dates of birth
One competitor registered for a recent competition with date of birth 2018-08-10.  I did not see any infants competing, so I think that was a mistake :)

Selecting the current year rather than your birth year is a pretty common mistake, and should be prevented by the account creation system.  Maybe we should require new accounts to be at least 3 years old, and say "if this is not a mistake please contact the WST" or something like that.

### PR DESCRIPTION
**Pull request by danayi13**
Fixes #3455

### CODE REVIEW

File: WcaOnRails/spec/models/user_spec.rb [hunk 1]
```diff
@@ -645,4 +645,18 @@
       expect(senior_delegate.editable_fields_of_user(user).to_a).to include(:delegate_status, :senior_delegate_id, :region)
     end
   end
+
+  describe "birthdate validations" do
+    it "requires birthdate in past" do
+      user = FactoryBot.create :user
```
Comment (jonatanklosko) [inline WcaOnRails/spec/models/user_spec.rb]: This appears twice and is a perfect candidate to make use of `let`, like so:
```ruby
describe "birthdate validations" do
  let(:user) { FactoryBot.create :user }

  it "requires birthdate in past" do
    user.dob = 5.days.from_now
    expect(user).to be_invalid_with_errors(dob: ["must be in the past"])
  end
  
   it "requires user over two years old" do
    user.dob = 5.days.ago
    expect(user).to be_invalid_with_errors(dob: ["must be at least two years old"])
  end
end
```
What [`let`](https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let) really gives us, is the ability to move such variable definition out of a single test and reuse it across many of those (while providing each test with a fresh value). Usually it removes repetition and perhaps make the tests itself a bit cleaner.
In cases like this it's somehow a matter of preference though, so I don't feel to strongly about this =)

**Comment by jonatanklosko** [comment]
@pedrosino certainly not, you can use `validate` with either a method name or a block. It's a matter of preference which one you go for. If you define the method right after `validate` call, using the name is kinda redundant, and in this case the name would be either long or not really descriptive ^^

**Comment by viroulep** [comment]
I'm not sure from whom we are waiting some input to merge this, so I'll ping @danayi13 just in case :)

Dana, do you plan to adjust your code to use `let` as Jonatan suggested or are you happy with how it looks now?
If you are I'll merge merge and deploy this!

**Comment by danayi13** [comment]
Hi Philippe,
I haven't had the time recently to do another paired programming
session/learn more about the code base (even though I really really want
to), so I haven't changed it. However, Jonatan's suggestion does make sense
and I think it makes the code a little cleaner. If you wouldn't mind, can
you please merge and deploy it?

Thank you so much,
Dana

**Comment by viroulep** [comment]
>If you wouldn't mind, can
you please merge and deploy it?

Of course!

Congrats for your first PR merged :tada:

**Pull request merged**